caluma/conftest.py

```diff
@@ -377,8 +378,37 @@ def custom_snapshot(*args):
 
 
 @pytest.fixture
-def at_date(freezer):
-    """Run a given piece of code at a certain date.
+def set_date(freezer):
+    """Run a block of code at a certain date (context manager).
+
+    Uses freezegun, but allows you to override the frozen date.
+
+    Provides a context manager, where everything within the "with"
+    block is run at the given date:
+
+    >>> @pytest.mark.freeze_time('2022-02-22')
+    >>> def test_foo(set_date):
+            assert datetime.now() == datetime(2022,2,22)
+
+            with set_date('2022-01-01'):
+                assert datetime.now() == datetime(2022,1,1)
+
+            assert datetime.now() == datetime(2022,2,22)
+    """
+
+    @contextlib.contextmanager
+    def make_context(date):
+        old_now = datetime.datetime.now()
```

Comment:
This might cause a runtime warning because it's a naive date, no?

Reply:
No, those dates are not used in Django`DateTimeField`s. It's only used to set a fake time during testing

Reply:
:+1: